hw1/02-palindrome.js

```diff
@@ -1,3 +1,50 @@
-const elem = document.querySelector('input');
+const elem = document.querySelector("input");
+const palindromeText = document.getElementById("palindrome");
 
-elem.addEventListener('input', handleInput);
+const isPalindrome = (num) => {
+  let l = 0;
+  let r = num.length - 1;
+
+  while (l <= r) {
+    if (num[l] !== num[r]) {
+      return false;
+    }
+    l += 1;
+    r -= 1;
+  }
+
+  return true;
+};
+
+const handleInput = () => {
+  const { value } = elem;
+  palindromeText.innerText = "";
+  palindromeText.classList.remove("text-success", "text-danger");
+
+  if (value.length === 0) {
+    palindromeText.innerText = "";
+    return;
+  }
+
+  if (Number.isNaN(value) || value.includes(" ")) {
+    palindromeText.innerText = "Non-numeric characters detected.";
+    palindromeText.classList.add("text-danger");
+    return;
+  }
```

Comment:
I could total miss it but can you give an example input that will prompt this text?

Reply:
The mdn docs mention that it is theoretically possible or something like that, but I couldn't reproduce it:
[" Logically, you should not be able to enter characters inside a number input other than numbers. Some browsers allow invalid characters, others do not; "](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number) 
I agree though that this is probably over doing it.